### PR TITLE
Use `MathOptInterface` in place of `MathProgBase` in repeated_game.jl

### LINKED ISSUE #109: Use `MathOptInterface` in place of `MathProgBase`
* [x] `normal_form_game.jl`
* [ ] `repeated_game.jl`

### PR DESCRIPTION
**Pull request by Yuya-Furusawa**
Close #109

### CODE REVIEW

File: src/normal_form_game.jl [hunk 1]
```diff
@@ -1,8 +1,5 @@
 #=
 Tools for normal form games.
-
-Authors: Daisuke Oyama
```
Comment (oyamad) [inline src/normal_form_game.jl]: Do this in another PR.

File: src/repeated_game.jl [hunk 2]
```diff
@@ -254,30 +254,54 @@ Given a constraint w ∈ W, this finds the worst possible payoff for agent i.
    here `nH` is the number of subgradients.
 - `C::Vector{Float64}` : The array containing the hyperplane levels.
 - `i::Int` : The player of interest.
-- `lp_solver::AbstractMathProgSolver` : Allows users to choose a particular
-  solver for linear programming problems. Options include ClpSolver(),
-  CbcSolver(), GLPKSolverLP() and GurobiSolver(). By default, it choooses
-  ClpSolver().
+- `lp_solver::Union{MathOptInterface.AbstractOptimizer,Function}` : Linear
```
Comment (oyamad) [inline src/repeated_game.jl]: This should be

```suggestion
- `lp_solver::Union{Type{<:MathOptInterface.AbstractOptimizer},Function}` : Linear
```

(The same for the docstring for `is_dominated`.)

File: src/repeated_game.jl [hunk 3]
```diff
@@ -254,30 +254,54 @@ Given a constraint w ∈ W, this finds the worst possible payoff for agent i.
...
+   programming solver to be used internally. Pass a
+   `MathOptInterface.AbstractOptimizer` type (such as `Clp.Optimizer`) if no
+   option is needed, or a function (such as `() -> Clp.Optimizer(LogLevel=0)`)
+   to supply options.
```
Comment (oyamad) [inline src/repeated_game.jl]: Mind the indentation.

File: src/repeated_game.jl [hunk 4]
```diff
@@ -254,30 +254,54 @@ Given a constraint w ∈ W, this finds the worst possible payoff for agent i.
...
 - `out::Float64` : Worst possible payoff for player i.
 """
-function worst_value_i(rpd::RepGame2, H::Matrix{Float64},
-                       C::Vector{Float64}, i::Int,
-                       lp_solver::MathProgBase.AbstractMathProgSolver=
-                       ClpSolver())
+function worst_value_i(
+        rpd::RepGame2, H::Matrix{Float64},
+        C::Vector{Float64}, i::Int,
+        lp_solver::Union{Type{TO},Function}=() -> Clp.Optimizer(LogLevel=0)
```
Comment (oyamad) [inline src/repeated_game.jl]: Indentation.

File: src/repeated_game.jl [hunk 5]
```diff
@@ -254,30 +254,54 @@ Given a constraint w ∈ W, this finds the worst possible payoff for agent i.
...
+    MOI.set(optimizer,
+            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x), 0.0))
+    MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+
+    # Add constraints
+    for i in 1:size(H,1)
+        MOI.add_constraint(
+            optimizer,
+            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(H[i,:],x), 0.0),
```
Comment (oyamad) [inline src/repeated_game.jl]: White space:

```suggestion
            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(H[i, :],x), 0.0),
```

File: src/repeated_game.jl [hunk 6]
```diff
@@ -287,12 +311,13 @@ end
 
 "See `worst_value_i` for documentation"
 worst_value_1(rpd::RepGame2, H::Matrix{Float64}, C::Vector{Float64},
-              lp_solver::MathProgBase.AbstractMathProgSolver=ClpSolver()) =
-    worst_value_i(rpd, H, C, 1, lp_solver)
+        lp_solver::Union{Type{TO},Function}=() -> Clp.Optimizer(LogLevel=0)
```
Comment (oyamad) [inline src/repeated_game.jl]: Indent.

File: src/repeated_game.jl [hunk 7]
```diff
@@ -287,12 +311,13 @@ end
...
+    ) where {TO<:MOI.AbstractOptimizer} = worst_value_i(rpd, H, C, 1, lp_solver)
+
 "See `worst_value_i` for documentation"
 worst_value_2(rpd::RepGame2, H::Matrix{Float64}, C::Vector{Float64},
-              lp_solver::MathProgBase.AbstractMathProgSolver=ClpSolver()) =
-    worst_value_i(rpd, H, C, 2, lp_solver)
+        lp_solver::Union{Type{TO},Function}=() -> Clp.Optimizer(LogLevel=0)
```
Comment (oyamad) [inline src/repeated_game.jl]: Indent.

File: src/repeated_game.jl [hunk 8]
```diff
@@ -321,10 +346,11 @@ hyperplane approximation described by Judd, Yeltekin, Conklin (2002).
   the geometry computations.
   (See [Polyhedra.jl](https://github.com/JuliaPolyhedra/Polyhedra.jl)
   docs for more info). By default, it chooses to use `Polyhedra.DefaultLibrary`.
-- `lp_solver::AbstractMathProgSolver` : Allows users to choose a particular
-  solver for linear programming problems. Options include ClpSolver(),
-  CbcSolver(), GLPKSolverLP() and GurobiSolver(). By default, it choooses
-  ClpSolver().
+- `lp_solver::Union{MathOptInterface.AbstractOptimizer,Function}` : Linear
```
Comment (oyamad) [inline src/repeated_game.jl]:
```suggestion
- `lp_solver::Union{Type{<:MathOptInterface.AbstractOptimizer},Function}` : Linear
```

File: src/repeated_game.jl [hunk 9]
```diff
@@ -321,10 +346,11 @@ hyperplane approximation described by Judd, Yeltekin, Conklin (2002).
...
+   programming solver to be used internally. Pass a
+   `MathOptInterface.AbstractOptimizer` type (such as `Clp.Optimizer`) if no
+   option is needed, or a function (such as `() -> Clp.Optimizer(LogLevel=0)`)
+   to supply options.
```
Comment (oyamad) [inline src/repeated_game.jl]: Indent.

File: src/repeated_game.jl [hunk 10]
```diff
@@ -403,11 +425,34 @@ function outerapproximation(
...
+                    MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+                    MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x), 0.0)
+                    )
+                MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+
+                # Add constraints
+                for i in 1:size(A,1)
+                    MOI.add_constraint(optimizer,
+                                       MOI.ScalarAffineFunction(
+                                        MOI.ScalarAffineTerm.(A[i,:],x), 0.0),
```
Comment (oyamad) [inline src/repeated_game.jl]:
```suggestion
                                       MOI.ScalarAffineTerm.(A[i, :],x), 0.0),
```

**Comment by oyamad** [comment]
Try running `outerapproximation` with `plib=CDDLib.Library()`.

**Comment by Yuya-Furusawa** [comment]
Sorry, I misunderstood. I will fix it and push it again.
Also, 

> Try running `outerapproximation` with `plib=CDDLib.Library()`.

Does not work and I got the error, `ERROR: type CDDGeneratorMatrix has no field V`.
I will also try to fix this issue.
